### PR TITLE
Implement a function to generate CVAT token and fix Credentials class

### PR DESCRIPTION
**Pull request by ptelang**
This commit adds a new function that authenticates a user against CVAT server and writes the authentication token to a file. The Credentials class is refactored and modified to work on Linux, Mac and Windows operating systems.

### CODE REVIEW

**Comment by ptelang** [comment]
Hi @courtneyambrozic, Can you please review this PR? Thanks.

**Comment by kybowm** [comment]
@ptelang, do you mind if I make docstring suggestions here? It's easier to do in a code review than in a separate commit.

**Comment by ptelang** [comment]
> @ptelang, do you mind if I make docstring suggestions here? It's easier to do in a code review than in a separate commit.

@kybowm, Yes, that works.

**Review by courtneyambrozic — COMMENTED** [review]
This looks great to me, Pankaj! I'll wait for Kyle's suggestions before approving.

File: cvpy/annotation/cvat/CVATAuthenticator.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    Defines a class with methods for CVAT authentication.
+    '''
+
+    # Max attempts for the user to provide valid URL, username and password before exiting
+    MAX_ATTEMPTS = 3
+
+    @staticmethod
+    def generate_cvat_token():
+        '''
+        Authenticates a user against a CVAT server. This function prompts the user to enter a CVAT server URL,
```
[the remark this answers is not in the document]
Comment (ptelang) [reply cvpy/annotation/cvat/CVATAuthenticator.py]: Hi @kybowm, I addressed your comments. Can you please review? Thanks.

**Review by kybowm — APPROVED** [review]
Looks good to me. Thanks, @ptelang!

**Review by courtneyambrozic — APPROVED** [review]
Looks good, Pankaj! Thanks!

**Pull request merged**